src/strgen/strgen_base.cpp

```diff
@@ -980,3 +982,113 @@ void LanguageWriter::WriteLang(const StringData &data)
 		}
 	}
 }
+
+static const std::string_view NBSP_TOKEN = "{NBSP}";
+
+static std::string ReplaceNBSP(std::string string)
+{
+	for (;;) {
+		auto iter = string.find(NBSP_TOKEN);
+		if (iter == std::string::npos) break;
+
+		string.replace(iter, NBSP_TOKEN.size(), NBSP);
+	}
+	return string;
+}
+
+/**
+ * Parse the \c NumberFormatSeparators out of the given format string, with the expected number of digits.
+ *
+ * Different cultures have different ways to separate their numbers when they get really big. In the Western world
+ * these are often called thousands separators which come every three digits counted from the back. The actual
+ * separator differs per country. In Chinese, Japanese and Korean they add a character every four digits counted
```

Comment:
```suggestion
 * separator differs per language. In Chinese, Japanese and Korean they add a character every four digits counted
```

Reply:
In South Africa the decimal separators are different than in the UK, yet both use English as language. See e.g. https://en.wikipedia.org/wiki/Decimal_separator. So it's quite debatable whether it's language or country.